src/pdiagmat.jl

```diff
@@ -74,63 +74,35 @@ LinearAlgebra.sqrt(a::PDiagMat) = PDiagMat(map(sqrt, a.diag))
 
 ### whiten and unwhiten
 
-function whiten!(r::StridedVector, a::PDiagMat, x::StridedVector)
-    n = dim(a)
-    @check_argdims length(r) == length(x) == n
-    v = a.diag
-    for i = 1:n
-        r[i] = x[i] / sqrt(v[i])
-    end
-    return r
-end
-
-function unwhiten!(r::StridedVector, a::PDiagMat, x::StridedVector)
-    n = dim(a)
-    @check_argdims length(r) == length(x) == n
-    v = a.diag
-    for i = 1:n
-        r[i] = x[i] * sqrt(v[i])
-    end
-    return r
-end
-
-function whiten!(r::StridedMatrix, a::PDiagMat, x::StridedMatrix)
-    r .= x ./ sqrt.(a.diag)
-    return r
-end
-
-function unwhiten!(r::StridedMatrix, a::PDiagMat, x::StridedMatrix)
-    r .= x .* sqrt.(a.diag)
-    return r
-end
-
+whiten!(r::AbstractVecOrMat, a::PDiagMat, x::AbstractVecOrMat) = r .= x ./ sqrt.(a.diag)
+unwhiten!(r::AbstractVecOrMat, a::PDiagMat, x::AbstractVecOrMat) = r .= x .* sqrt.(a.diag)
 
 ### quadratic forms
 
 quad(a::PDiagMat, x::AbstractVector) = wsumsq(a.diag, x)
 invquad(a::PDiagMat, x::AbstractVector) = invwsumsq(a.diag, x)
 
-function quad!(r::AbstractArray, a::PDiagMat, x::StridedMatrix)
+function quad!(r::AbstractArray, a::PDiagMat, x::AbstractMatrix)
     m, n = size(x)
     ad = a.diag
     @check_argdims m == length(ad) && length(r) == n
-    @inbounds for j = 1:n
+    @inbounds for j in axes(x, 2)
```

Comment:
This is not correct I think: We need indices for r but also indices of x along the second dimension. They don't have to match, and in particular r might be of any shape currently. So I think we want something like
```suggestion
    @inbounds for (ir, j) in zip(eachindex(r), axes(x, 2))
```

Reply:
This is correct as long as I also check the axes (rather than the length) above in the `checkdims`.
It isn't maximally general.
But such generality i think better to come in a follow up PR that does a lot more testing.
Much easier to screw up.

Reply:
fixed check (just noting here so will push shortly)